### PR TITLE
Error handling WIP

### PR DESCRIPTION
**Pull request by JamesAPetts**
CC @dannyrb #1598

Requires these two updates:
https://github.com/dcmjs-org/dicomweb-client/pull/25
https://github.com/cornerstonejs/cornerstoneWADOImageLoader/pull/313

The following redirects to `https://ohif.org/` as a test on failed network requests:
The user can check `error.status` to get the status code. `error.request` holds the original `XMLHttpRequest` and `error.response` is the server's response, if any.

The user could configure a modal to pop open here, as I have given the callback function access to the `servicesManager`.

```
window.config = function(props) {
  var servicesManager = props.servicesManager;

  return {
    routerBasename: '/',
    enableGoogleCloudAdapter: true,
    enableGoogleCloudAdapterUI: false,
    showStudyList: true,
    httpErrorHandler: error => {
      debugger;

      // This is 429 when rejected from the public idc sandbox too often.
      console.warn(error.status);

      // Could use services manager here to bring up a dialog/modal if needed.
      console.warn('test, navigate to https://ohif.org/');
      window.location = 'https://ohif.org/';
    },
    healthcareApiEndpoint: 'https://idc-sandbox-002.appspot.com/v1beta1',
  };
};
```

### CODE REVIEW

File: platform/viewer/public/config/idc.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      debugger;
+
+      // This is 429 when rejected from the public idc sandbox too often.
+      console.warn(error.status);
+
+      // Could use services manager here to bring up a dialog/modal if needed.
+      console.warn('test, navigate to https://ohif.org/');
+      window.location = 'https://ohif.org/';
+    },
+    healthcareApiEndpoint: 'https://idc-sandbox-002.appspot.com/v1beta1',
```
Comment (dannyrb) [inline platform/viewer/public/config/idc.js:17]: I realize this has probably existed forever, but not sure any of the google cloud keys are documented

File: platform/viewer/src/App.js [hunk 2]
```diff
@@ -31,7 +31,8 @@ import {
 import i18n from '@ohif/i18n';
 
 // TODO: This should not be here
-import './config';
+//import './config';
+import { setConfiguration } from './config';
```
Comment (dannyrb) [inline platform/viewer/src/App.js:35]: Appreciate that you changed this to a module.

File: platform/viewer/public/config/idc.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+window.config = function(props) {
+  var servicesManager = props.servicesManager;
+
+  return {
+    routerBasename: '/',
+    enableGoogleCloudAdapter: true,
+    enableGoogleCloudAdapterUI: false,
+    showStudyList: true,
+    httpErrorHandler: error => {
```
Comment (dannyrb) [inline platform/viewer/public/config/idc.js:9]: Do both clients return the same error shape here? 🤔

File: docs/latest/viewer/configuration.md [hunk 4]
```diff
@@ -30,6 +38,32 @@ window.config = {
...
+    const { request: xhr, response, status } = err;
+    const { responseType, statusText } = xhr;
+
+    // In local files, status is 0 upon success in Firefox
+    if (xhr.readyState === XMLHttpRequest.DONE) {
+      console.log(statusText, response, responseType);
+    } else {
+      console.warn('Likely CORS error');
+    }
+  },
```
Comment (dannyrb) [inline docs/latest/viewer/configuration.md:66]: Called out props in doc

File: platform/core/package.json [hunk 5]
```diff
@@ -33,14 +33,14 @@
...
-    "cornerstone-wado-image-loader": "^3.0.0",
+    "cornerstone-wado-image-loader": "^3.1.0",
     "dicom-parser": "^1.8.3"
   },
   "dependencies": {
     "@babel/runtime": "^7.5.5",
     "ajv": "^6.10.0",
     "dcmjs": "^0.12.0",
-    "dicomweb-client": "^0.5.2",
+    "dicomweb-client": "^0.6.0",
```
Comment (dannyrb) [inline platform/core/package.json:43]: Bumping versions to make sure we can leverage the new hook/options

File: platform/viewer/public/config/idc.js (hunk 3 above)
Comment (dannyrb) [reply platform/viewer/public/config/idc.js:9]: Verified: yes. Marking resolved.

File: docs/latest/viewer/configuration.md (hunk 4 above)
Comment (JamesAPetts) [reply docs/latest/viewer/configuration.md:66]: Champion 💯

**Pull request merged**